src/pages/ExcelPage.tsx

```diff
@@ -0,0 +1,83 @@
+/**
+ * ExcelPage.tsx
+ * Example of uploading an excel spreadsheet into DB and displaying the data
+
+
+*/
+import { useState, useEffect } from 'react';
+import { v4 as uuid } from 'uuid';
+
+// material-ui
+import { Box, Typography } from '@mui/material';
+import StaffTable from '../sections/staff/StaffTable';
+import MainCard from '../components/MainCard';
+import { Staff, staffService } from '../sections/staff/staffService';
+
+// sheetJS
+import { read, utils } from "xlsx";
+
+
+const ExcelPage = () => {
+
+    const [uploadMsg, setUploadMsg] = useState('');
+
+    const [staff, setStaff] = useState<Staff[]>([]);
+
+    useEffect(() => {
+        staffService.getStaff()
+            .then(res => setStaff(res))
+    }, [])
+
+    const handleUpload = async (event: React.ChangeEvent<HTMLInputElement>) => {
+        if (!event.target.files) return;
+        const file = event.target.files[0];
+        // get parsed array from the uploaded excel file
+        const newStaffData = await handleParse(file)
+        // upload parsed data to supabase
+        await staffService.postStaff(newStaffData)
```

Comment:
error handling?

Reply:
Agreed. I decided to add a try/catch inside `handleParse`, and have the `postStaff` throw an error, which gets caught. 

So now an error gets thrown if something happens at the parsing stage (like a column is missing), and the upload stage (supabase rejects the upload because of a unique constraint)

Let me know what you think, error handling is definitely an area I'd like to improve in

Reply:
Handling bulk object creation has many options, so choosing how to handle errors depends on application requirements.  For example, if one commit fails then all commits fail.

For this application, let's display an error message if there are any parsing exceptions and do not commit.   Bonus points for telling the user what/where the error is.  If there are partial commit errors, display message for what was committed and what failed.

FYI, for applications that I've written in the past, there would be a web page representation of the file with results added via various techniques (e.g. color-coding, popups, results column)